src/main/java/seedu/address/logic/commands/AddListingCommand.java

```diff
@@ -76,29 +82,69 @@ public AddListingCommand(Name listingName, Price price, Area area, Address addre
         this.buyers = buyers;
     }
 
+    // needs SLAP
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
         List<Person> lastShownList = model.getFilteredPersonList();
 
         Person seller = model.getPersonByName(this.seller);
+
         if (!lastShownList.contains(seller)) {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_INPUT);
         }
+
+        if (seller instanceof Buyer) {
+            throw new CommandException(MESSAGE_NOT_SELLER);
+        }
+
         Set<Person> personBuyers = new HashSet<>();
-        for (Name b: buyers) {
-            Person buyer = model.getPersonByName(b);
-            if (!lastShownList.contains(buyer)) {
-                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_INPUT);
+
+        if (!isNull(buyers)) {
```

Comment:
If you did not run an instanceof check, does this mean that you can accidentally add sellers as buyers of the listing? This could be an unintended outcome.

Reply:
You could also add a MESSAGE_NOT_BUYER as a static field and run instanceof again.

Reply:
Resolved. Discussed with members. 

Intended outcome. Sellers can buy listings but buyers cannot sell listings